### PR TITLE
Inventory/Bank: changed upgrades and infusions from array to object

### PR DESCRIPTION
**Pull request by codemasher**
For consistencies sake. This way we can grab all item IDs by recursively searching for the key "id" or for example use functions like [array_column()](http://php.net/manual/function.array-column.php).

### CODE REVIEW

**Comment by lye** [comment]
I'm not sure how much I like this; it really depends on whether or not you consider applied upgrades to be "item stacks", which is how most other things are encoded. They can neither stack nor have upgrades/infusions applied to them, which makes me feel that they don't need to use the item stack encoding.

That, and I'm a bit concerned about breaking backwards compatibility.

**Comment by FrederikNS** [comment]
I was working on something using the item api, and this was probably the
one thing that seemed most weird.

Switching to an object is a good idea in my opinion.

**Comment by darthmaim** [comment]
> I'm not sure how much I like this; it really depends on whether or not you consider applied upgrades to be "item stacks", which is how most other things are encoded. They can neither stack nor have upgrades/infusions applied to them, which makes me feel that they don't need to use the item stack encoding.
> That, and I'm a bit concerned about breaking backwards compatibility

Haha, yesterday I started to write exactly the same comment, but somehow forgot to finish and send it.

I wouldn't break backwards compatibility for this, I'm also not seeing how `$upgradeIds = array_column($itemStack->upgrades, 'ids')` is any simpler than just `$upgradeIds = $itemStack->upgrades`.

TL;DR: :-1:

**Comment by codemasher** [comment]
So after a conversation with darthmaim we agreed that just "id" or just an array of numbers isn't the best thing since there are different kind of IDs in the response. This way we're still not sure which kind of ID whe have. Cliff said once that the API should be self explaining - using "item_id", "skin_id" or "whatever_id" would be exactly that.
